packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_poolswap.test.ts

```diff
@@ -186,14 +181,70 @@ describe('dex.poolswap()', () => {
       fromAmount: new BigNumber('2'),
       toScript: script,
       toTokenId: pairs.CAT.tokenB,
-      maxPrice: {
-        integer: new BigNumber('0'),
-        fraction: new BigNumber('0.3')
-      }
+      maxPrice: new BigNumber('0.00000003')
     }, script)
 
     const promise = sendTransaction(container, txn)
     await expect(promise).rejects.toThrow(DeFiDRpcError)
     await expect(promise).rejects.toThrow('Price is higher than indicated')
   })
+
+  it('should pass with 5.0', async () => {
+    providers.randomizeEllipticPair()
+    await container.waitForWalletBalanceGTE(1)
+    const addressLP = await container.getNewAddress()
+    await addPoolLiquidity(container, {
+      tokenA: 'DFI',
+      amountA: 100,
+      tokenB: 'BIRD',
+      amountB: 20,
+      shareAddress: addressLP
+    })
+    await providers.setupMocks()
+    await utxosToAccount(container, 100, { address: await providers.getAddress() })
+    await sendTokensToAddress(container, await providers.getAddress(), 2, 'BIRD')
+    await fundEllipticPair(container, providers.ellipticPair, 10)
+    const script = await providers.elliptic.script()
+
+    const txn = await builder.dex.poolSwap({
+      fromScript: script,
+      fromTokenId: pairs.BIRD.tokenA,
+      fromAmount: new BigNumber('2'),
+      toScript: script,
+      toTokenId: pairs.BIRD.tokenB,
+      maxPrice: new BigNumber('5.0')
+    }, script)
+    const promise = sendTransaction(container, txn)
+    await expect(promise).resolves.not.toThrow()
+  })
+
+  it('should fail with 4.99999999', async () => {
+    providers.randomizeEllipticPair()
+    await container.waitForWalletBalanceGTE(1)
+    const addressLP = await container.getNewAddress()
+    await addPoolLiquidity(container, {
+      tokenA: 'DFI',
+      amountA: 100,
+      tokenB: 'BIRD',
```

Comment:
use another token for this, otherwise, your test scenario is populated as swap will cause pool reverse change causing max price behavior to change as well